### PR TITLE
Add create method with metrics parameter in ConnectionProvider (#2313)

### PR DESCRIPTION
**Pull request by tuguri8**
Add create method with name, maxConnections, metricsEnabled parameters.

Related to https://github.com/reactor/reactor-netty/issues/2313

### CODE REVIEW

File: reactor-netty-core/src/main/java/reactor/netty/resources/ConnectionProvider.java [hunk 1]
```diff
@@ -161,6 +161,28 @@ static ConnectionProvider create(String name, int maxConnections) {
...
+	 * Further connections will be pending acquisition until {@link #DEFAULT_POOL_ACQUIRE_TIMEOUT}
+	 * and the default pending acquisition max count will be 2 * max connections value.
+	 *
+	 * @param name the connection pool name
+	 * @param maxConnections the maximum number of connections before starting pending
+	 * acquisition on existing ones
+	 * @param metricsEnabled true enables metrics collection; false disables it
+	 *
+	 * @return a new {@link ConnectionProvider} to cache and reuse a fixed maximum
+	 * number of {@link Connection}
```
Comment (violetagg) [inline reactor-netty-core/src/main/java/reactor/netty/resources/ConnectionProvider.java:177]: Please add

`* @since 1.0.21`

**Review by violetagg — CHANGES_REQUESTED** [review]
@tuguri8 Thanks for the PR. Only one small addition that provides information in which version this method was added.

**Comment by violetagg** [comment]
@tuguri8 We need to exclude this new static method from the compatibility check. Please add to `build.gradle`

```
diff --git a/reactor-netty-core/build.gradle b/reactor-netty-core/build.gradle
index 951caf774..67913b06e 100644
--- a/reactor-netty-core/build.gradle
+++ b/reactor-netty-core/build.gradle
@@ -194,6 +194,9 @@ task japicmp(type: JapicmpTask) {
 	ignoreMissingClasses = true
 	includeSynthetic = true
 	onlyIf { "$compatibleVersion" != "SKIP" }
+	methodExcludes = [
+			"reactor.netty.resources.ConnectionProvider#create(java.lang.String, int, boolean)"
+	]
 }
 
 tasks.japicmp.dependsOn(downloadBaseline)
```

File: reactor-netty-core/src/main/java/reactor/netty/resources/ConnectionProvider.java (hunk 1 above)
Comment (tuguri8) [reply reactor-netty-core/src/main/java/reactor/netty/resources/ConnectionProvider.java:177]: added since tag

**Review by violetagg — APPROVED** [review]
@reactor/netty-team PTAL

**Review by pderop — APPROVED** [review]
looks good to me

**Comment by violetagg** [comment]
@pderop Thanks for the review

**Pull request merged**